src/m3ninx/index/segment/fst/segment.go

```diff
@@ -330,7 +320,7 @@ func (r *fsSegment) MatchRegexp(field []byte, regexp []byte, compiled index.Comp
 	var (
 		fstCloser     = x.NewSafeCloser(termsFST)
 		pl            = r.opts.PostingsListPool().Get()
-		iter, iterErr = termsFST.Search(re, minByteKey, nil)
+		iter, iterErr = termsFST.Search(re, compiled.PrefixBegin, compiled.PrefixEnd)
```

Comment:
If prefixBegin and prefixEnd are both nil (as can happen with call to `LiteralPrefix(...)`) should `minByteKey` still be specified here?

Reply:
So it doesn't need to be, `nil` will cause more scans than `minByteKey`

Yep! Hopefully people aren't doing `.*`, but beyond that it'll help a lot for any reasonable query. For the '.*' case, i'm going to add more data to the segment (postings list per unique field) sometime down the road.